Handreiking API Design/Handreiking APi Design.md

```diff
@@ -54,14 +63,21 @@ De naamgevingsconventie is nl. als volgt:
 
 Voorbeelden hiervan zijn:
 
-* `ZaakContactmoment`
-* `VerzoekInformatieobject`
-* `ZaakBesluit`
-* `BesluitInformatieobject`
+* `zaakverzoek`
+* `zaakcontactmoment`
+* `verzoekinformatieobject`
+* `zaakbesluit`
+* `besluitinformatieobject`
 
 In een gegevensmodel kunnen in dit geval 2 klasses opgenomen zijn met dezelfde
 naam maar elk opgenomen in een ander domein.
 
```

Comment:
Volgens mij klopt dit niet. De relatieklasse in het dominatie domein is sterk getypeerd. Bijvoorbeeld ZaakInformatieobject, ZaakVerzoek, ZaakBesluit etc. 
Terwijl in het niet-dominante domein sprake is van ObjectInformatieobject, ObjectVerzoek, ObjectBesluit etc. In een dergelijke meer generieke klasse is in eerste instantie alleen te zien dat bijvoorbeeld een Informatieobject gekoppeld is aan "iets anders" maar wat dat is zie je nog niet. Een koppeling van een Informatieobject met zowel een Zaak als met een Besluit levert dus twee ObjectInformatieobjecten op.

Reply:
Wat klopt er dan niet? Kun je iets precieser zijn?

Reply:
Daarbij is dit te plek om aan te geven dat alleen d sterk getypeerde klasse (met eigenschappen) in de API wordt opgenomen die ontsloten wordt. Nattuurrlijk wordt er ook een endpoint voor alleen de koppeltable gedefinieerd, maar die wordt alleen door de beheerder van de relatieklasse gebruikt.

Reply:
@michielverhoef, zoals hierboven als verzocht, graag een bevestiging. Kan jij ook een antwoord geven op mijn vraag hierboven?

Reply:
> 
> 
> Wat klopt er dan niet? Kun je iets precieser zijn?

Er zijn geen twee klasses met dezelfde naam. Bijvoorbeeld in de ZRC bestaat ZaakInformatieobject, diezelfde relatie (zonder extra informatie, die is alleen in de ZRC opgeslagen) heet in de DRC ObjectInformatieobject.

Het leggen van de ObjectInformatieobject relatie wordt ook ontsloten via een API (in dit geval de DRC API) alleen wordt via autorisatie geregeld dat alleen de eigenaar van de relatie (in dit geval de ZRC) dit kan.

Reply:
Hmm, ik weet niet zeker of ik @michielverhoef begrijp maar...

> In een gegevensmodel kunnen in dit geval 2 klasses opgenomen zijn met dezelfde naam maar elk opgenomen in een ander domein.

Deze zin klopt niet of ik lees hem verkeerd (hoewel dubbeling wellicht kan voorkomen om redenen). We hanteren de suggestie van Henri al grotendeels, dus het lijkt me sterk er 2 x bijv. ``ZaakVerzoek`` als resource ergens bestaat?

Daarnaast bestaan deze niet resources niet:

* `verzoekinformatieobject`
* `besluitinformatieobject`

Deze noemen we (in het DRC): `objectinformatieobject` omdat het een relatie van *een* `object` naar een `informatieobject` betreft, waarbij `object` een `verzoek` of `besluit` (of `zaak`) kan zijn.

In retrospect zou `objectinformatieobject` wellicht `informatieobjectobject` moeten heten maar ja... dat leest ook wel vervelend.

Reply:
Als ik de zin

> In een gegevensmodel kunnen in dit geval 2 klasses opgenomen zijn met dezelfde naam maar elk opgenomen in een ander domein.

wijzig in

> In een gegevensmodel kunnen om bepaalde redenen 2 klasses opgenomen zijn met dezelfde naam maar elk opgenomen in een ander domein. Over het algemeen zal echter dat wat beschreven staat in de volgende sectie (Relatieklassen met meerdere objecttypen) gelden.

is dat dan voldoende? @Joeri, je geeft tenminste aan dat dubbelingen wel voor zouden kunnen komen.

Reply:
Als dat (er kunnen dubbele namen voorkomen) het hoofdpunt is wat je wilt maken, vond ik de eerste versie ook prima. Lijkt me echter onwenselijk, dus als het voorkomen kan worden is dat beter :)

Reply:
Blijkbaar was mijn intentie bij die versie niet helemaal duidelijk dus is het goed om dat beter te beschrijven. Je opmerking dat het onwenselijk is meenemend stel ik voor dat we het beschrijven als:

> In een gegevensmodel kunnen om bepaalde redenen 2 klasses opgenomen zijn met dezelfde naam maar elk opgenomen in een ander domein. Dit is echter geen ideale situatie welke als het even kan voorkomen moet worden. Over het algemeen zal ook dat wat beschreven staat in de volgende sectie (Relatieklassen met meerdere objecttypen) gelden.

Reply:
Prima.